### PR TITLE
Update allowed values for `Repository` collaborator permissions

### PR DESCRIPTION
**Pull request by flying-sheep**
`add_to_collaborators` accepts a few more

### CODE REVIEW

**Comment by flying-sheep** [comment]
Hi, this is still a valid change.

File: github/Repository.py [hunk 1]
```diff
@@ -815,7 +815,7 @@ def add_to_collaborators(self, collaborator, permission=github.GithubObject.NotS
         """
         :calls: `PUT /repos/{owner}/{repo}/collaborators/{user} <https://docs.github.com/en/rest/reference/repos#collaborators>`_
```
Comment (EnricoMi) [inline github/Repository.py:816]:
```suggestion
        :calls: `PUT /repos/{owner}/{repo}/collaborators/{user} <https://docs.github.com/en/rest/collaborators/collaborators#add-a-repository-collaborator>`_
```

**Review by EnricoMi — APPROVED** [review]

**Comment by EnricoMi** [comment]
Please resolve conflicts and I am happy to merge. Thanks for the contribution!

**Comment by flying-sheep** [comment]
done!

File: github/Repository.py [hunk 2]
```diff
@@ -1036,8 +1036,10 @@ def web_commit_signoff_required(self) -> bool:
...
         """
-        :calls: `PUT /repos/{owner}/{repo}/collaborators/{user} <https://docs.github.com/en/rest/reference/repos#collaborators>`_
-        :param permission: string 'pull', 'push' or 'admin'
+        :calls: `PUT /repos/{owner}/{repo}/collaborators/{user} <https://docs.github.com/en/rest/collaborators/collaborators#add-a-repository-collaborator>`_
+        :param collaborator: string or :class:`github.NamedUser.NamedUser`
+        :param permission: string 'pull', 'push', 'admin', 'maintain', or 'triage'
+        :rtype: None
         """
         assert isinstance(collaborator, github.NamedUser.NamedUser) or isinstance(collaborator, str), collaborator
         assert is_optional(permission, str), permission
```
Comment (EnricoMi) [inline github/Repository.py:1045]:
```suggestion
        assert permission in ["pull", "push", "admin", "maintain", "triage", NotSet], permission
```

Comment (EnricoMi) [reply github/Repository.py]: I completely missed that the `permission` values have never been asserted, fixed.

**Pull request merged**